apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/environment/PortalMetaDomainConsts.java

```diff
@@ -0,0 +1,198 @@
+package com.ctrip.framework.apollo.portal.environment;
+
+import com.ctrip.framework.apollo.core.utils.ApolloThreadFactory;
+import com.ctrip.framework.apollo.core.utils.NetUtil;
+import com.ctrip.framework.apollo.tracer.Tracer;
+import com.ctrip.framework.apollo.tracer.spi.Transaction;
+import com.ctrip.framework.foundation.internals.ServiceBootstrap;
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * a fake class only use in apollo-portal
+ * For the replace of com.ctrip.framework.apollo.core.MetaDomainConsts in apollo-core
+ * @see com.ctrip.framework.apollo.core.MetaDomainConsts
+ * @author wxq
+ */
+public class PortalMetaDomainConsts {
+
+    public static final String DEFAULT_META_URL = "http://apollo.meta";
+
+    // env -> meta server address cache
+    private static final Map<Env, String> metaServerAddressCache = Maps.newConcurrentMap();
+    private static volatile List<PortalMetaServerProvider> metaServerProviders = null;
+
+    private static final long REFRESH_INTERVAL_IN_SECOND = 60;// 1 min
+    private static final Logger logger = LoggerFactory.getLogger(PortalMetaDomainConsts.class);
+    // comma separated meta server address -> selected single meta server address cache
+    private static final Map<String, String> selectedMetaServerAddressCache = Maps.newConcurrentMap();
+    private static final AtomicBoolean periodicRefreshStarted = new AtomicBoolean(false);
+
+    private static final Object LOCK = new Object();
+
+    /**
+     * Return one meta server address. If multiple meta server addresses are configured, will select one.
+     */
+    public static String getDomain(Env env) {
+        String metaServerAddress = getMetaServerAddress(env);
+        // if there is more than one address, need to select one
+        if (metaServerAddress.contains(",")) {
+            return selectMetaServerAddress(metaServerAddress);
+        }
+        return metaServerAddress;
+    }
+
+    /**
+     * Return meta server address. If multiple meta server addresses are configured, will return the comma separated string.
+     */
+    public static String getMetaServerAddress(Env env) {
+        if (!metaServerAddressCache.containsKey(env)) {
+            initMetaServerAddress(env);
+        }
+
+        return metaServerAddressCache.get(env);
+    }
+
+    private static void initMetaServerAddress(Env env) {
+        if (metaServerProviders == null) {
```

Comment:
I think we don't need this  meta server provider spi for portal, so we could simplify this logic.

Reply:
Initialize `metaServerProviders` without lazy? i.e initial it when class loaded? (use `final` to the field)

Reply:
> don't need this meta server provider spi for portal

It mean that there are no need to use the `com.ctrip.framework.apollo.portal.environment.PortalMetaServerProvider`, simply cancel this interface, and use `com.ctrip.framework.apollo.portal.environment.PortalLegacyMetaServerProvider`  directly?

Reply:
I think we could remove the PortalMetaServerProvider spi for now

Reply:
Ok. I will remove the interface